src/S3.ts

```diff
@@ -42,11 +42,11 @@ class S3 {
         });
     }
 
-    public async uploadBlob(key: string, data: any, eTag: string) {
+    public async uploadBlob(key: string, data: any, eTag: string, version: string) {
         const params = {
             Bucket: this.bucketName,
             Key: key,
-            Metadata: {etag: eTag },
+            Metadata: {etag: eTag , version},
```

Comment:
This is a key value system, so `{etag: eTag , version: version}`

Reply:
lint and codefactor were showing issue with it that's why I did it this way

Reply:
and js support if we write same name for key and value we can just write it like this

Reply:
ok...